pymc_marketing/mmm/components/base.py

```diff
@@ -223,13 +231,159 @@ def _create_distributions(self, dim_name: str) -> dict[str, pt.TensorVariable]:
+
+        """
+        with pm.Model():
+            self._create_distributions()
+            return pm.sample_prior_predictive(**sample_prior_predictive_kwargs).prior
+
+    def plot_curve(
+        self,
+        curve: xr.DataArray,
+        color: str = "C0",
+        ax: plt.Axes | None = None,
+        sample_kwargs: dict | None = None,
+        hdi_kwargs: dict | None = None,
+    ) -> plt.Axes:
+        """Plot curve HDI and samples.
+
+        Parameters
+        ----------
+        curve : xr.DataArray
+            The curve to plot.
+        color : str, optional
+            The color of the curve. Defaults to "C0".
+        ax : plt.Axes, optional
+            The axes to plot on. Defaults to None.
+        sample_kwargs : dict, optional
+            Keyword arguments for the plot_curve_sample function. Defaults to None.
+        hdi_kwargs : dict, optional
+            Keyword arguments for the plot_curve_hdi function. Defaults to None.
+
+        Returns
+        -------
+        plt.Axes
+            The axes with the plot.
+
+        """
+        hdi_kwargs = hdi_kwargs or {}
+        sample_kwargs = sample_kwargs or {}
+        ax = self.plot_curve_hdi(curve, color=color, ax=ax, **hdi_kwargs)
+        ax = self.plot_curve_sample(curve, color=color, ax=ax, **sample_kwargs)
+        return ax
+
+    def plot_curve_sample(
+        self,
+        curve: xr.DataArray,
+        color: str = "C0",
+        ax: plt.Axes | None = None,
+        n: int = 10,
+        rng: np.random.Generator | None = None,
+        plot_kwargs: dict | None = None,
+    ) -> plt.Axes:
+        """Plot samples from the curve.
+
+        Parameters
+        ----------
+        curve : xr.DataArray
+            The curve to plot.
+        color : str, optional
+            The color of the curve. Defaults to "C0".
+        ax : plt.Axes, optional
+            The axes to plot on. Defaults to None.
+        n : int, optional
+            The number of samples to plot. Defaults to 10.
+        rng : np.random.Generator, optional
+            The random number generator to use. Defaults to None.
+        plot_kwargs : dict, optional
+            Keyword arguments for the plot function. Defaults to None.
+
+        Returns
+        -------
+        plt.Axes
+            The axes with the plot.
+
+        """
+        df_curve = curve.to_series().unstack()
+
+        df_sample = df_curve.sample(n=n, random_state=rng)
+
+        ax = ax or plt.gca()
+        plot_kwargs = plot_kwargs or {}
+        plot_kwargs["color"] = color
+        plot_kwargs["alpha"] = plot_kwargs.get("alpha", 0.3)
+        plot_kwargs["legend"] = False
+        df_sample.T.plot(ax=ax, **plot_kwargs)
+
+        return ax
+
+    def plot_curve_hdi(
+        self,
+        curve: xr.DataArray,
+        color: str = "C0",
+        ax: plt.Axes | None = None,
+        hdi_kwargs: dict | None = None,
+        plot_kwargs: dict | None = None,
+    ) -> plt.Axes:
```

Comment:
We usually return the figure... shall we keep that convention (no strong opinion)

Reply:
ok! No strong preference. Just think about it from a user perspective :)